### ISSUE TITLE
Cannot use existing database

### ISSUE THREAD

**Issue by TomTheBear** [influxdata/influxdb#8501]
### Bug report

__System info:__
InfluxDB: 1.2.1-1
InfluxDB shell version: 1.2.1
OS: Ubuntu 15.10 with kernel 4.2.0-42-generic

__Steps to reproduce:__

1. Create around 9990 databases (in my case 9996)
2. Create a new database using the InfluxDB shell `influx`
```
> create database test2
```
3. Try to use it:
```
> use test2
ERR: Database test2 doesn't exist. Run SHOW DATABASES for a list of existing databases.
> show databases
[...]
testdb
test2

> 
```
It works for databases created earlier.

__Expected behavior:__ Let me use the newly created database

__Actual behavior:__ Server answers that no such database exists

__Additional info:__
Log just contains: `[I] 2017-06-16T14:57:58Z SHOW DATABASES service=query`

**Comment by rbetts** [comment]
Using 10k databases isn't going to be a great experience regardless. Data is stored on disk per database and this many databases will create very bad I/O patterns. 

Regardless, you should prefer 1.2.4 on the 1.2.x release branch vs. 1.2.1 (https://portal.influxdata.com/downloads).

**Comment by TomTheBear** [comment]
Thanks for the comment but this does _not_ help at all. Refusing bugs (wontfix) that don't exist in the intended use-case is not good style.
Moreover, when using 10k databases is a bad idea, why does the system allow to create so many and does not refuse the 'create database' command above a reasonable threshold? And much worse, why does it allow to create them but you cannot use them? Besides you cannot 'use' them in the shell, also writing to it over the HTTP API returns 404. Where is it documented that above X databases, you can create more but not use them?
Is this bug fixed in 1.2.4? Or is it just a general text template that is added for anyone who doesn't use the newest version?
